.github/workflows/validate-theme-entry.yml

```diff
@@ -0,0 +1,263 @@
+                }
+
+                if (themes.filter(t => t.name === theme.name).length > 1) {
+                    addError('There is already a theme with this name');
+                }
+
+                if (themes.filter(t => t.repo === theme.repo).length > 1) {
+                    addError('There is already a entry pointing to the `' + theme.repo + '` repository');
+                }
+
+                // we can't validate further, as there is no repo set
+                if(!theme.hasOwnProperty('repo')) {
+                    return;
+                }
+
+                // Validate theme repo
+                let repoInfo = theme.repo.split('/');
+                if (repoInfo.length !== 2) {
+                    addError(`It seems like you made a typo in the repository field ${theme.repo}`);
+                    return;
+                }
+
+                let [owner, repo] = repoInfo;
+                console.log(`Repo info: ${owner}/${repo}`);
+
+                if (owner.toLowerCase() !== author.toLowerCase()) {
+                    try {
+                        const isInOrg = await github.rest.orgs.checkMembershipForUser({ org: owner, username: author });
+                        if (!isInOrg) {
+                            throw undefined;
+                        }
+                    } catch (e) {
+                        addError(`The newly added entry is not at the end, or you are submitting on someone else's behalf. The last plugin in the list is: \`${plugin.repo}\`.`);
+                    }
+                }
+                try {
+                    const repository = await github.rest.repos.get({ owner, repo });
+                    if (!repository.data.has_issues) {
+                        addWarning('Your repository does not have issues enabled.');
+                    }
+                } catch (e) {
+                    addError(`It seems like you made a typo in the repository field ${theme.repo}`);
+                    return;
+                }
+
+                let manifest = {};
+
+                try {
+                    let manifestFile = await github.rest.repos.getContent({
+                        owner,
+                        repo,
+                        path: 'manifest.json',
+                    });
+
+                    manifest = JSON.parse(Buffer.from(manifestFile.data.content, 'base64').toString('utf-8'));
+                } catch (e) {
+                    addError(`You don't have a valid \`manifest.json\` at the root of your repo.`);
+                    return;
+                }
+
+                let validManifestKeys = ['name', 'minAppVersion', 'author', 'authorUrl'];
+                for (let key in manifest) {
+                    if (!validKeys.includes(key) && !key === 'version') {
+                        addError(`Your manifest has the invalid \`${key}\` property.`);
+                    }
+                }
+
+                for (let key in validManifestKeys) {
```

Comment:
Again, don't think we need this.

Reply:
This check will stop the Action to fail without a message.
The checks after this might fail accessing `undefined` fields, the only message will be in the action logs.

Reply:
In that case we should rename the var to `requiredManifestKeys`.